### DISCUSSION TITLE
Viem Bug: createClient type error

### DISCUSSION THREAD

**Discussion by 00MB** [wevm/wagmi#2771] [General]
./node_modules/viem/dist/types/clients/createClient.d.ts:32:14
Type error: Type parameter declaration expected.

  30 | };
  31 | export type Client<transport extends Transport = Transport, chain extends Chain | undefined = Chain | undefined, account extends Account | undefined = Account | undefined, rpcSchema extends RpcSchema | undefined = undefined, extended extends Extended | undefined = Extended | undefined> = Client_Base<transport, chain, account, rpcSchema> & (extended extends Extended ? extended : unknown) & {
> 32 |     extend: <const client extends Extended>(fn: (client: Client<transport, chain, account, rpcSchema, extended>) => client) => Client<transport, chain, account, rpcSchema, Prettify<client> & (extended extends Extended ? extended : unknown)>;
     |              ^
  33 | };
  34 | type Client_Base<transport extends Transport = Transport, chain extends Chain | undefined = Chain | undefined, account extends Account | undefined = Account | undefined, rpcSchema extends RpcSchema | undefined = undefined> = {
  35 |     /** The Account of the Client. */
error Command failed with exit code 1.

Viem version ^1.0.0
"wagmi": "^1.3.9",
"@wagmi/core": "^1.3.8",
"phantom-wagmi-connector": "^1.0.7",

**Comment by attilagaliba** [comment]
same

**Reply by DorianDeSimone** [reply]
same here, someone can help ?

**Reply by DorianDeSimone** [reply]
Found !
The problem is the Typescript version, in the `viem` docs it's said 

> Types currently require using TypeScript v5.0.4 or greater.

I was using 4.9.4 in my project, i needed to update the Typescript version.

Hope it will help !
